src/wallet/wallet.h

```diff
@@ -250,6 +250,12 @@ class CWallet final : public WalletStorage, public interfaces::Chain::Notificati
     //! the current wallet version: clients below this version are not able to load the wallet
     int nWalletVersion GUARDED_BY(cs_wallet){FEATURE_BASE};
 
+    //! The extended public key to be used for new automatically generated descriptors
+    CExtPubKey m_xpub GUARDED_BY(cs_wallet);
+    //! The previous and current extended keys that are used for automatically generated descriptors
+    std::map<CExtPubKey, CKey> m_hd_keys GUARDED_BY(cs_wallet);
```

Comment:
This presupposes that we will and should allow changing active HD key for a given wallet. I'd like to better understand for what use-cases this is needed.

Reply:
I'm not entirely sure of the reasoning, but it's been requested as required feature.